config/initializers/rack_attack.rb

```diff
@@ -105,6 +105,10 @@ def paging_request?
     req.authenticated_user_id if (req.post? && req.path.match?(API_DELETE_REBLOG_REGEX)) || (req.delete? && req.path.match?(API_DELETE_STATUS_REGEX))
   end
 
+  throttle('throttle_oauth_application_registrations/ip', limit: 5, period: 30.minutes) do |req|
+    req.throttleable_remote_ip if req.post? && req.path == '/api/v1/apps'
+  end
```

Comment:
Rate-limiting this seems sensible, but looking into mastodon.social's logs, we are seeing a few patterns with more than 5 registrations per IP per 30 minutes.

None seem ill-intentioned, but the most aggressive look like clients or apps that re-register with each use. I don't think rate-limiting them is necessarily an issue as I think those should clients/scripts should probably be fixed instead.

Some are more difficult to explain, like a few IPs seem to regularly re-register Mastodon for Android. The bursts are usually below the proposed rate-limit, but I think it may make sense to account for someone wanting to try out multiple apps or re-install an app to debug issues, in which case a rate-limit would probably make the situation more confusing than helpful for them.

Therefore I'd make this a little less strict just to err on the safe side:
```suggestion
  throttle('throttle_oauth_application_registrations/ip', limit: 5, period: 10.minutes) do |req|
    req.throttleable_remote_ip if req.post? && req.path == '/api/v1/apps'
  end
```

Reply:
Sure!